### PR TITLE
#156882172 Users should have a profile page

### PR DESCRIPTION
**Pull request by chuzksy-codeactive**
#### What does this PR do?
Users should have a profile page
#### Description of Task to be completed?
Users should be able to see their profile page and see their personal log of the previously played game and the number of rounds they have won
#### How should this be manually tested?
Simple run `npm run start`
Go to `http:localhost:4000/#!/profile`
#### What are the relevant pivotal tracker stories?
The endpoint should be authenticated
#### Screenshot
screen shot 2018-05-15 at 1 31 30 pm

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -351,3 +362,15 @@ exports.invitePlayersByMail = function (req, res) {
...
   });
 };
+
+/** Gathers User Info and Games Logs */
+exports.game = function (req, res) {
+  Game.find().exec((err, games) => {
+    if (err) return err;
+    if (!games) return new Error('Failed to load Game');
+    return res.status(200).json({
+      data: games,
```
Comment (andela-moe) [inline app/controllers/users.js:372]: Is this returning all the games or just the games for the user that is logged in?

File: config/routes.js [hunk 2]
```diff
@@ -103,4 +136,6 @@ module.exports = function (app, passport) {
     validator.gameValidation,
     saveGame
   );
+
+  app.get('/api/usergames', users.game);
```
Comment (andela-moe) [inline config/routes.js:140]: Based on the convention prescribed, `usergames` in this endpoint should be in `camelCase`.

File: public/js/controllers/index.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ angular.module('mean.system').controller('IndexController', [
         successResponse => {
           $scope.global.authenticated = true;
           $scope.global.user = successResponse[0].data.data.user;
+          console.log($scope.global.user);
```
Comment (andela-moe) [inline public/js/controllers/index.js:43]: Do we need this `console.log($scope.global.user);` here?

File: public/js/controllers/index.js [hunk 4]
```diff
@@ -165,5 +166,49 @@ angular.module('mean.system').controller('IndexController', [
...
+    $scope.userGames = () => {
+      useGames();
+    };
+
+    const useGames = () => {
+      let userGameDetails = [];
+      if (window.user !== null) {
+        $http.get('/api/usergames').then((res) => {
+          if (res.data.code === 200) {
+            console.log(res.data.data);
```
Comment (andela-moe) [inline public/js/controllers/index.js:183]: Is this `console.log(res.data.data);` relevant here?

File: test/user/profile.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      .get('/api/usergames')
+      .expect(200)
+      .end((req, res) => {
+        expect(res.body).to.have.property('data');
+        expect(res.body).to.have.property('code');
+        expect(res.status).to.equal(200);
+        done();
+      });
+  });
+});
```
Comment (andela-moe) [inline test/user/profile.test.js:19]: Your describe block suggests you that this test suite is for this endpoint `/api/profile` However, you seem to have made a `get` request to `/api/usergames`. Please verify what endpoint this test is valid for.

Comment (andela-moe) [reply test/user/profile.test.js:19]: The language used here `Should return 200 on get profile call` does not see to communicate what you are testing for. Once you respond to the feedback about you should have more clarity on what this should have been framed as.

Comment (andela-moe) [reply test/user/profile.test.js:19]: I believe you should have been able to provide more test specs that prove you have considered other edge cases. E.g. the user has not played a single game, what is returned in that case?

**Review by andela-moe — COMMENTED** [review]
Please review comments
